### DISCUSSION TITLE
RingBuf overflow?

### DISCUSSION THREAD

**Discussion by XunChangqing** [bluenviron/gortsplib#182] [Questions]
In Push method of RingBuf, it didn't check the size of buf?
Is it possible that the buf overflow?

Thanks

**Comment by aler9** [comment]
Hello, the ring buffer can't overflow but a new entry can override an older entry.

When a ring buffer is full, we're left with two choices:

1. discard incoming packets. Since we're dealing with real-time video streams, this would mean pausing the video and losing new frames.
2. discard old packets and allow new packets to override them, preserving real-time.

The 2nd option in my opinion is the better one and that's the reason why there are no size checks.

**Comment by nu774** [comment]
We can at least increase the ring buffer size via Server.WriteBufferCount.
However, as I've written in https://github.com/bluenviron/gortsplib/issues/101, ring buffer for RTCP feedback direction is fixed at 8 and is not modifiable from user.
Therefore, if we are manually pushing too much packets into that direction via ServerSession.WritePacketRTP() or ServerSession.WritePacketRTCP(), the ring buffer will easily be full, yielding packet drops or out of order delivery.
